src/weapons/weapon.js

```diff
@@ -0,0 +1,26 @@
+import WeaponPlugin from 'phaser3-weapon-plugin';
+
+export class Weapon {
```

Comment:
Maybe it's better to extend the Weapon class that's defined in the WeaponPlugin.

Reply:
Or maybe we use a different pattern.  WeaponPlugin provides lots of convenience methods for customising the weapon.  It's easy to change things like spread, fire rate, bullet icon, physics.  Maybe we just use some factories that produce weapons.